python/cudf/cudf/core/index.py

```diff
@@ -1563,11 +1564,39 @@ def __contains__(self, item):
         else:
             return False
 
-    def copy(self, deep=True):
+    def copy(self, deep=True, name=None, dtype=None):
         """
         Make a copy of this object.
+
+        Parameters
+        ----------
+        name : object optional (default: None), name of index
+        deep : Bool (default: True)
+            Ignored for RangeIndex
+        dtype : numpy dtype (default: None)
+            Target dtype for underlying range data
+
+        Returns
+        -------
+        New RangeIndex instance with same range, casted to new dtype
         """
-        return RangeIndex(start=self._start, stop=self._stop, name=self.name)
+        if name is None and dtype is None:
+            return RangeIndex(
+                start=self._start, stop=self._stop, name=self.name
+            )
+
+        dtype = self.dtype if dtype is None else dtype
+
+        if not is_range_compatible_dtype(dtype):
+            raise ValueError(f"Expected Signed Integer Type, Got {dtype}")
+
+        name = self.name if name is None else name
+
+        _idx_new = as_index(
+            self._values.copy(deep=deep).astype(dtype), name=name
+        )
+
+        return _idx_new
```

Comment:
This looks like incorrect behavior. In Pandas even with specifying a `dtype`, `name`, and `deep=True` I'm getting a `RangeIndex` in return.

Reply:
Yes. I'm seeing as long as dtype is SignedInteger type, all return is `RangeIndex`, otherwise it raises.

Also would like to check about the understanding of `deep` copy for `RangeIndex`. I think pandas simply ignores the `deep` flag. My reasoning is: `RangeIndex` does not store underlying `_column` object. So deep and shallow copy does not make a difference. In some situation, `_values` may be called to instantiate a `_column` object, but it should not be considered as the data for `RangeIndex`.

Reply:
Yea, deep doesn't really change the behavior of `RangeIndex.copy` as far as I can tell.